### PR TITLE
enable DistriOptimizerV2

### PR DESCRIPTION
**Pull request by Le-Zheng**
move to new PR

### CODE REVIEW

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/models/Topology.scala [hunk 1]
```diff
@@ -97,13 +98,27 @@ abstract class KerasNet[T](implicit val tag: ClassTag[T], implicit val ev: Tenso
           ds = local,
           criterion = this.criterion)
       case distriDataSet: DistributedDataSet[MiniBatch[T]] =>
-        new InternalDistriOptimizer(_model = this,
-          _dataset = distriDataSet,
-          _criterion = this.criterion)
+        EngineRef.getOptimizerVersion() match {
+          case OptimizerV1 =>
+            new InternalDistriOptimizer(_model = this,
```
Comment (qiuxin2012) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/models/Topology.scala:103]: move to a uniform apply function

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/models/Topology.scala [hunk 2]
```diff
@@ -1508,6 +1539,176 @@ private[zoo] class InternalDistriOptimizer[T: ClassTag] (
...
+
+  override def evaluate(
+        validationSet: FeatureSet[MiniBatch[T]],
+        validationMethod: Array[ValidationMethod[T]]
+      ): Map[ValidationMethod[T], ValidationResult] = {
+    val validateRDD = validationSet.toDistributed().data(train = false)
+    val sc = validateRDD.sparkContext
+
+    val coresPerNode = EngineRef.getCoreNumber()
+    val _subModelNumber = EngineRef.getEngineType() match {
```
Comment (qiuxin2012) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/models/Topology.scala:1681]: unused variable?

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/models/Topology.scala (hunk 1 above)
Comment (Le-Zheng) [reply zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/models/Topology.scala:103]: Put apply function in an Object?

**Comment by Le-Zheng** [comment]
Jenkins passed related tests: http://10.239.47.210:18888/view/ZOO-PR/job/ZOO-PR-Validation/4128/

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/estimator/Estimator.scala [hunk 3]
```diff
@@ -145,16 +147,31 @@ class Estimator[T: ClassTag] private[zoo](
...
-            .setCheckpointDir(modelDir)
-            .setOptimMethods(optimMethods)
-            .setNumOfSlice(d.numOfSlice)
-          if ((logDir != null)  && (appName != null)) {
-            val trainSummary = TrainSummary(logDir, appName)
-            val valSummary = ValidationSummary(logDir, appName)
-            internalEstimator.asInstanceOf[Optimizer[_, _]]
-              .setTrainSummary(trainSummary)
-              .setValidationSummary(valSummary)
+          EngineRef.getOptimizerVersion() match {
```
Comment (qiuxin2012) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/estimator/Estimator.scala:150]: move `internalEstimator =` to here.
and move two 
```
 if ((logDir != null)  && (appName != null)) {
                val trainSummary = TrainSummary(logDir, appName)
                val valSummary = ValidationSummary(logDir, appName)
                internalEstimator.asInstanceOf[Optimizer[_, _]]
                  .setTrainSummary(trainSummary)
                  .setValidationSummary(valSummary)
              }
```
to one.

Comment (Le-Zheng) [reply zoo/src/main/scala/com/intel/analytics/zoo/pipeline/estimator/Estimator.scala:150]: updated in https://github.com/intel-analytics/analytics-zoo/pull/2809

**Pull request merged**